core/domain/prod_validation_jobs_one_off.py

```diff
@@ -40,13 +42,53 @@ def _external_id_relationships(cls):
             model to validate, and whose values are the corresponding model
             classes.
         """
-        raise NotImplementedError
+        return {}
+
+    @classmethod
+    def _custom_external_id_relationship(cls, item):  # pylint: disable=unused-argument
+        """Define custom external relationships.
+
+        Sometimes we need more complicated logic to determine which external
+        model to fetch. Two example use cases are:
+        1) the field name follows some convention e.g. {exp_id}.{exp_version}
+        2) the field name has to be extracted from a structured field
+
+        Args:
+            item: ndb.Model. Entity to validate.
+
+        Returns:
+            A list of tuples, each consists of a field name (for debug output),
```

Comment:
Please add typeinfo at the start, e.g. `list(tuple(str, str, list(str))).` (I think? Not sure.)